tests/shared/src/test/scala/cats/tests/TraverseSuite.scala

```diff
@@ -36,6 +36,19 @@ abstract class TraverseSuite[F[_]: Traverse](name: String)(implicit ArbFInt: Arb
     }
   }
 
+  test(s"Traverse[$name].mapAccumulate") {
+    forAll { (fa: F[Int], init: Int, fn: ((Int, Int)) => (Int, Int)) =>
+      val lhs = fa.mapAccumulate(init)((s, a) => fn((s, a)))
+
+      val rhs = fa.foldLeft((init, List.empty[Int])) { case ((s1, acc), a) =>
+        val (s2, b) = fn((s1, a))
+        (s2, b :: acc)
+      }
+
+      assert(lhs._2.toList === rhs._2.reverse)
```

Comment:
Since `mapAccumulate` returns both `S` and `F[B]` then it makes sense to add a check for `S` here as well, IMO.